lib/tests/v22_test.go

```diff
@@ -217,53 +226,44 @@ func manifestEqual(manifest, expected *schema.ImageManifest) error {
 	if manifest.ACKind != expected.ACKind {
 		return fmt.Errorf("expected ACKind %q, got %q", expected.ACKind, manifest.ACKind)
 	}
+
 	if manifest.ACVersion != expected.ACVersion {
 		return fmt.Errorf("expected ACVersion %q, got %q", expected.ACVersion, manifest.ACVersion)
 	}
+
 	if !reflect.DeepEqual(*manifest.App, *expected.App) {
 		return fmt.Errorf("expected App %v, got %v", *expected.App, *manifest.App)
 	}
-	for _, label := range []string{"arch", "os", "version"} {
-		if err := checkLabel(label, manifest, expected); err != nil {
-			return err
-		}
-	}
-	for _, ann := range []string{
-		"author",
-		"created",
-		"appc.io/docker/repository",
-		"appc.io/docker/entrypoint",
-		"appc.io/docker/cmd",
-	} {
-		if err := checkAnnotation(ann, manifest, expected); err != nil {
-			return err
-		}
-	}
 
-	return nil
-}
-
-func checkLabel(name string, manifest, expected *schema.ImageManifest) error {
-	got, ok := manifest.GetLabel(name)
-	if !ok {
-		return fmt.Errorf("missing %q label", name)
-	}
-	exp, _ := expected.GetLabel(name)
-	if got != exp {
-		return fmt.Errorf("expected label %q, got %q", exp, got)
+	if len(manifest.Labels) != len(expected.Labels) {
+		return fmt.Errorf("Labels not equal: %v != %v", manifest.Labels, expected.Labels)
 	}
 
-	return nil
-}
+	for _, label := range manifest.Labels {
+		el, ok := expected.Labels.Get(label.Name.String())
+		if !ok {
+			return fmt.Errorf("expected label %v to exist, did not", label.Name)
+		}
+		if label.Value != el {
+			return fmt.Errorf("expected label %v values to match, but %v != %v", label.Name, el, label.Value)
+		}
+	}
 
-func checkAnnotation(name string, manifest, expected *schema.ImageManifest) error {
-	got, ok := manifest.GetAnnotation(name)
-	if !ok {
-		return fmt.Errorf("missing %q annotation", name)
+	if len(manifest.Annotations) != len(expected.Annotations) {
+		return fmt.Errorf("annotations not equal: %v != %v", manifest.Annotations, expected.Annotations)
 	}
-	exp, _ := expected.GetAnnotation(name)
-	if got != exp {
-		return fmt.Errorf("expected annotation %q, got %q", exp, got)
+	for _, ann := range manifest.Annotations {
+		ea, ok := expected.Annotations.Get(ann.Name.String())
+		if ea == "variant" {
+			// Unknown value, always okay
```

Comment:
Before, it only asserted on a subset of values (which didn't include `appc.io/docker/imageid`). However, the test image generated doesn't appear to be idempotent so this varies per test.

The expected value is set to this arbitrary value to skip asserting on it.